spec/rubocop/cop/style/array_first_last_spec.rb

```diff
@@ -58,4 +58,92 @@
       arr[0][-1]
     RUBY
   end
+
+  it 'registers an offense when using `arr.[](0)`' do
+    expect_offense(<<~RUBY)
+      arr.[](0)
+          ^^^^^ Use `first`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      arr.first
+    RUBY
+  end
+
+  it 'registers an offense when using `arr&.[](0)`' do
+    expect_offense(<<~RUBY)
+      arr&.[](0)
+           ^^^^^ Use `first`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      arr&.first
+    RUBY
+  end
+
+  it 'registers an offense when using `arr.[] 0`' do
+    expect_offense(<<~RUBY)
+      arr.[] 0
+          ^^^^ Use `first`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      arr.first
+    RUBY
+  end
```

Comment:
@koic unlike #13658, I think this is a valid offense/autocorrect, but please let me know if I'm overlooking something 😄

Reply:
Yeah, this is fine as there are no semantic changes. Thank you!